httpMappingHandler.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/Sirupsen/logrus"
+	"io/ioutil"
+	"net/http"
+	trans "github.com/Financial-Times/transactionid-utils-go"
+	"time"
+)
+
+type ErrorMessage struct {
+	Message string `json:"message"`
+}
+
+func newHTTPErrorMessage(msg string) ErrorMessage {
+	return ErrorMessage{Message: msg}
+}
+
+type HTTPMappingHandler interface {
+	handle(w http.ResponseWriter, r *http.Request)
+}
+
+type defaultHTTPMappingHandler struct {
+	messageToNativeMapper MessageToNativeMapper
+	imageSetMapper        ImageSetMapper
+}
+
+func newHTTPMappingHandler(messageToNativeMapper MessageToNativeMapper, imageSetMapper ImageSetMapper) HTTPMappingHandler {
+	return defaultHTTPMappingHandler{
+		messageToNativeMapper: messageToNativeMapper,
+		imageSetMapper:        imageSetMapper,
+	}
+}
+
+func (h defaultHTTPMappingHandler) handle(w http.ResponseWriter, r *http.Request) {
+	tid := trans.GetTransactionIDFromRequest(r)
+
+	body, err := ioutil.ReadAll(r.Body)
+	defer h.closeResponseBody(r)
+	if err != nil {
+		h.warnAndWriteToHTTP500(fmt.Sprintf("Cound't read from request body. %v\n", err), w)
+		return
+	}
+
+	native, err := h.messageToNativeMapper.Map(body)
+	if err != nil {
+		h.writeToHTTP500(fmt.Sprintf("Error mapping native message. %v\n", err), w)
+		return
+	}
+
+	imageSets, err := h.imageSetMapper.Map(native, time.Now().Format(uppDateFormat), tid)
+	if err != nil {
+		h.writeToHTTP500(fmt.Sprintf("Error mapping the given content. %v\n", err), w)
+		return
+	}
+
+	marshaledJSONImageSets, err := json.Marshal(imageSets)
+	if err != nil {
+		h.warnAndWriteToHTTP500(fmt.Sprintf("Couldn't marshall built-up image-sets to JSON. %v\n", err), w)
+		return
+	}
+
+	w.Header().Add("Content-Type", "application/json;charset=utf-8")
```

Comment:
i added to the beginning of the function, all the possible responses are in json.